great_expectations/expectations/registry.py

```diff
@@ -340,10 +340,7 @@ def get_metric_function_type(
     metric_name: str, execution_engine: ExecutionEngine
 ) -> Optional[Union[MetricPartialFunctionTypes, MetricFunctionTypes]]:
     try:
-        metric_definition = _registered_metrics[metric_name]
-        provider_fn, _provider_class = metric_definition["providers"][
-            type(execution_engine).__name__
-        ]
+        provider_fn, _provider_class = get_metric_provider(metric_name, execution_engine)
```

Comment:
This is a cleanup to use get_metric_provider instead of looking into the registry directly. There are a few other places in this file we reach into the registry directly but we never use the provider there. Ideally, this registry would be a class so we couldn't grab the whole metric definition and use an undefined metric provider but that's a bigger refactor that we can do in a different PR.